pallets/creditcoin/src/migrations/mod.rs

```diff
@@ -44,3 +44,34 @@ pub(crate) fn migrate<T: Config>() -> Weight {
 
 	weight
 }
+
+macro_rules! projection_alias {
+	($($config: path : { $($typ: ident),+ }),+) => {
+		$(
+			$(
+				paste::paste! {
+					type [<$typ Of>]<T> = <T as $config>::$typ;
+				}
+			)+
+		)+
+	};
+}
```

Comment:
Just a comment, the average case doesn't seem to be any more concise.

Reply:
Fair enough lol

Reply:
To be on the same page, I meant for example `T::Hash` vs `Hashfor<T>` which is the case for the majority of the simple non-ambiguous associated types.